### PR TITLE
Ensure the word "marker" is used accurately in docs, add linter to examples 

### LINKED ISSUE #1781: Run a javascript linter on the examples
It is important to have examples that are idiomatic and correct. We should find a linter capable of giving the ~~html,~~ javascript ~~and css~~ a once-over before we publish. Right now this is all done by hand and, as an imperfect human, I make mistakes :grin:

### LINKED ISSUE #2809: Rename "ElementMarker" to "Marker", ensure terminology is used accurately in docs
- **Markers** are HTML elements fixed to a geographic coordinate on the map. Markers are styled and made interactive with the full power of HTML, CSS, and Javascript. Users should be able to work with about a hundred markers on a map.
 - **Symbol layers** are small icons & text drawn within WebGL fixed to a geographic coordinate on the map. Symbols are styled and made interactive with the GL style spec and GL JS APIs. Users should be able to work with many thousands of symbols on a map.

 - [x] rename `ElementMarker` to `Marker` in the codebase
 - [x] ensure all references to "markers" in GL JS API docs are accurate
 - [x] ensure all references to "markers" in GL JS examples are accurate
 - [x] ensure all references to "markers" in /help docs are accurate

cc @mourner @jfirebaugh @lyzidiamond @samanpwbb @tristen @1ec5

### PR DESCRIPTION
**Pull request by lucaswoj**
fixes #1781
fixes #2809 

# Todo

 -  [x] manually test all examples

### CODE REVIEW

File: docs/_posts/examples/3400-01-03-fitbounds.html [hunk 1]
```diff
@@ -31,13 +31,14 @@
     zoom: 9
 });
 
-function fit() {
+document.getElementById('fit').addEventListener('click', function() {
```
Comment (jfirebaugh) [inline docs/_posts/examples/3400-01-03-fitbounds.html]: This change looks incomplete. Did you mean to remove `onclick='fit();'` above?

File: docs/_posts/examples/3400-01-03-flyto-options.html [hunk 2]
```diff
@@ -35,7 +35,7 @@
 
 var isAtStart = true;
 
-function fly() {
+document.getElementById.addEventListener('click', function() {
```
Comment (jfirebaugh) [inline docs/_posts/examples/3400-01-03-flyto-options.html]: Also incomplete.

**Comment by jfirebaugh** [comment]
👍

File: docs/_posts/examples/3400-01-03-fitbounds.html (hunk 1 above)
Comment (lucaswoj) [inline docs/_posts/examples/3400-01-03-fitbounds.html]: Yes. I did. Thank you!

**Pull request merged**